### PR TITLE
Allow setting the limit on std::io::Take.

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
Fixes https://github.com/rust-lang/rust/issues/27269.

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Arguably this can also be not implemented since out-of-tree implementation should involve no more than `take.into_inner().take(new_limit)`, though this is more ergonomic (`&mut self` vs `self`).

**Comment by sfackler** [comment]
This seems reasonable to me - might be worth explicitly mentioning that it doesn't "reset" the count of what's been read?

@rfcbot fcp merge

**Comment by Mark-Simulacrum** [comment]
I'll fix the code if we decide to move forward with this, hopefully landing after https://github.com/rust-lang/rust/pull/42612.

**Comment by dtolnay** [comment]
@rfcbot reviewed

**Comment by alexcrichton** [comment]
ping @aturon / @brson on checkboxes

**Comment by Mark-Simulacrum** [comment]
@brson I believe this is only waiting on approval from you, will push a fix for the nit @sfackler noted with regards to docs.

**Comment by carols10cents** [comment]
ping @brson, still waiting for your checkbox here!

File: src/libstd/io/mod.rs [hunk 1]
```diff
@@ -1761,6 +1761,35 @@ impl<T> Take<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn limit(&self) -> u64 { self.limit }
 
+    /// Sets the number of bytes that can be read before this instance will
+    /// return EOF. This is the same as constructing a new `Take` instance, so
+    /// the amount of bytes read or the previous limit value do not affect the
```
Comment (aturon) [inline src/libstd/io/mod.rs]: "do not affect the effect" is a somewhat confusing phrasing. Also missing period.

**Comment by aturon** [comment]
@Mark-Simulacrum 

Left a nit, otherwise r=me

**Comment by Mark-Simulacrum** [comment]
Fixed the nit (I think). @brson hasn't checked off yet, so I'm hesitant to r+ without that.

**Comment by aturon** [comment]
@bors: r+

@Mark-Simulacrum for PRs (rather than RFCs), after a week if most folks have signed off we go forward.

**Pull request merged**